### DISCUSSION TITLE
Contentlayer doesn't work properly on Windows

### DISCUSSION THREAD

**Discussion by aIexnder** [community/community#109713] [Programming Help] [answered]
### Body

Even though I configured Contentlayer correctly, when I run the server, it doesn't generate the MDX content posts. However, when I deploy it to production, they are generated correctly and display properly. This is inconvenient because I can only preview the article after deploying it.

I would appreciate an explanation of how to resolve this issue on Windows.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by fraineralex** [comment, marked as answer]
I faced the same issue while developing my blog. I couldn't find information on resolving this problem on Windows, so what I did was create a specific branch where I added a route to render an array with all the articles. 

For example:

`app/dev/all-posts/page.tsx`
```tsx

import { allPosts } from 'contentlayer/generated'

export default function Page() {
  const postsJSON = JSON.stringify(allPosts, null, 2)

  return (
    <section className='text-zinc-100 px-2'>
      <pre>{postsJSON}</pre>
    </section>
  )
}
```

Once you do this, all you need to do is go to the deployment that Vercel makes from that branch, navigate to the route, copy the array of articles, and paste it into the `.contentlayer/generated/Post/_index.mjs` file.

```ts
export const allPosts = []  // <- replace the array
```

This way, you'll have all the articles as if your machine had generated them, allowing you to test styles and other configurations of your posts.

Hope this helps!

> [!NOTE]
> The content of the file in `.contentlayer/generated/Post/_index.mjs` will disappear every time the server is run.
